In [0]:
# ========================================
# 🎮 PROJET STEAM - CHARGEMENT DES DONNÉES
# ========================================

# Importer les fonctions PySpark nécessaires
from pyspark.sql import SparkSession
from pyspark.sql.functions import *

print("✅ Imports réussis !")

✅ Imports réussis !


In [0]:
# Charger le fichier JSON depuis le bucket S3
df = spark.read.json("s3://full-stack-bigdata-datasets/Big_Data/Project_Steam/steam_game_output.json")

print(f"✅ Données chargées avec succès !")
print(f"📊 Nombre total de lignes : {df.count()}")

✅ Données chargées avec succès !
📊 Nombre total de lignes : 55691


In [0]:
# Afficher le schéma complet (structure des colonnes)
print("📋 STRUCTURE DES DONNÉES :")
df.printSchema()

📋 STRUCTURE DES DONNÉES :
root
 |-- data: struct (nullable = true)
 |    |-- appid: long (nullable = true)
 |    |-- categories: array (nullable = true)
 |    |    |-- element: string (containsNull = true)
 |    |-- ccu: long (nullable = true)
 |    |-- developer: string (nullable = true)
 |    |-- discount: string (nullable = true)
 |    |-- genre: string (nullable = true)
 |    |-- header_image: string (nullable = true)
 |    |-- initialprice: string (nullable = true)
 |    |-- languages: string (nullable = true)
 |    |-- name: string (nullable = true)
 |    |-- negative: long (nullable = true)
 |    |-- owners: string (nullable = true)
 |    |-- platforms: struct (nullable = true)
 |    |    |-- linux: boolean (nullable = true)
 |    |    |-- mac: boolean (nullable = true)
 |    |    |-- windows: boolean (nullable = true)
 |    |-- positive: long (nullable = true)
 |    |-- price: string (nullable = true)
 |    |-- publisher: string (nullable = true)
 |    |-- release_date: string 

In [0]:
# Afficher les 10 premières lignes
display(df.limit(10))

data,id
"List(10, List(Multi-player, Valve Anti-Cheat enabled, Online PvP, Shared/Split Screen PvP, PvP), 13990, Valve, 0, Action, https://cdn.akamai.steamstatic.com/steam/apps/10/header.jpg?t=1666823513, 999, English, French, German, Italian, Spanish - Spain, Simplified Chinese, Traditional Chinese, Korean, Counter-Strike, 5199, 10,000,000 .. 20,000,000, List(true, true, true), 201215, 999, Valve, 2000/11/1, 0, Play the world's number 1 online action game. Engage in an incredibly realistic brand of terrorist warfare in this wildly popular team-based game. Ally with teammates to complete strategic missions. Take out enemy sites. Rescue hostages. Your role affects your team's success. Your team's success affects your role., List(266, 1191, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, 5426, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, 227, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, 2784, null, null, null, null, null, null, null, null, null, null, null, null, 1607, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, 4831, null, null, null, null, null, null, null, null, null, 1707, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, 632, null, null, null, null, null, null, null, null, null, null, null, 3392, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, 131, null, null, 769, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, 881, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, 289, null, null, null, 3353, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, 614, null, null, null, null, null, null, 304, null, null, null, 1344, null, null, 1864, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, 1192), game, )",10
"List(1000000, List(Single-player, Partial Controller Support, Steam Achievements, Steam Cloud), 0, IndigoBlue Game Studio, 0, Action, Adventure, Indie, https://cdn.akamai.steamstatic.com/steam/apps/1000000/header.jpg?t=1655723048, 999, English, Korean, Simplified Chinese, ASCENXION, 5, 0 .. 20,000, List(false, false, true), 27, 999, PsychoFlux Entertainment, 2021/05/14, 0, ASCENXION is a 2D shoot 'em up game where you explore the field to progress. Players must overcome puzzles, traps, elite units, boss fights, and other various obstacles while navigating the field. Grow stronger through rewards earned, 

In [0]:
# Compter le nombre de colonnes
nb_colonnes = len(df.columns)

# Compter le nombre de lignes
nb_lignes = df.count()

print("=" * 50)
print("📈 STATISTIQUES DU DATASET")
print("=" * 50)
print(f"Nombre de jeux (lignes) : {nb_lignes:,}")
print(f"Nombre de colonnes : {nb_colonnes}")
print("=" * 50)

# Afficher la liste des colonnes principales
print("\n📋 COLONNES DISPONIBLES :")
for i, col_name in enumerate(df.columns, 1):
    print(f"  {i}. {col_name}")

📈 STATISTIQUES DU DATASET
Nombre de jeux (lignes) : 55,691
Nombre de colonnes : 2

📋 COLONNES DISPONIBLES :
  1. data
  2. id


In [0]:
# ========================================
# 🔍 EXTRACTION DE LA STRUCTURE IMBRIQUÉE
# ========================================

# Créer un nouveau dataframe en extrayant tous les champs de "data"
df_clean = df.select(
    col("id"),
    col("data.appid").alias("appid"),
    col("data.name").alias("name"),
    col("data.developer").alias("developer"),
    col("data.publisher").alias("publisher"),
    col("data.positive").alias("positive"),
    col("data.negative").alias("negative"),
    col("data.owners").alias("owners"),
    col("data.price").alias("price"),
    col("data.initialprice").alias("initialprice"),
    col("data.discount").alias("discount"),
    col("data.ccu").alias("ccu"),
    col("data.languages").alias("languages"),
    col("data.genre").alias("genre"),
    col("data.categories").alias("categories"),
    col("data.platforms").alias("platforms"),
    col("data.header_image").alias("header_image")
)

print("✅ Données aplaties avec succès !")

✅ Données aplaties avec succès !


In [0]:
# Afficher les premières lignes du nouveau dataframe
print("👀 APERÇU DES DONNÉES APLATIES :")
print("=" * 60)

display(df_clean.limit(10))

👀 APERÇU DES DONNÉES APLATIES :


id,appid,name,developer,publisher,positive,negative,owners,price,initialprice,discount,ccu,languages,genre,categories,platforms,header_image
10,10,Counter-Strike,Valve,Valve,201215,5199,"10,000,000 .. 20,000,000",999,999,0,13990,"English, French, German, Italian, Spanish - Spain, Simplified Chinese, Traditional Chinese, Korean",Action,"List(Multi-player, Valve Anti-Cheat enabled, Online PvP, Shared/Split Screen PvP, PvP)","List(true, true, true)",https://cdn.akamai.steamstatic.com/steam/apps/10/header.jpg?t=1666823513
1000000,1000000,ASCENXION,IndigoBlue Game Studio,PsychoFlux Entertainment,27,5,"0 .. 20,000",999,999,0,0,"English, Korean, Simplified Chinese","Action, Adventure, Indie","List(Single-player, Partial Controller Support, Steam Achievements, Steam Cloud)","List(false, false, true)",https://cdn.akamai.steamstatic.com/steam/apps/1000000/header.jpg?t=1655723048
1000010,1000010,Crown Trick,NEXT Studios,"Team17, NEXT Studios",4032,646,"200,000 .. 500,000",599,1999,70,99,"Simplified Chinese, English, Japanese, Traditional Chinese, French, German, Spanish - Spain, Russian, Portuguese - Brazil","Adventure, Indie, RPG, Strategy","List(Single-player, Partial Controller Support, Steam Achievements, Steam Cloud, Steam Trading Cards)","List(false, false, true)",https://cdn.akamai.steamstatic.com/steam/apps/1000010/header.jpg?t=1655724189
1000030,1000030,"Cook, Serve, Delicious! 3?!",Vertigo Gaming Inc.,Vertigo Gaming Inc.,1575,115,"100,000 .. 200,000",1999,1999,0,76,English,"Action, Indie, Simulation, Strategy","List(Multi-player, Single-player, Co-op, Steam Achievements, Steam Cloud, Shared/Split Screen, Full controller support, Steam Trading Cards, Shared/Split Screen Co-op, Remote Play on Phone, Remote Play on Tablet, Remote Play on TV, Remote Play Together)","List(false, true, true)",https://cdn.akamai.steamstatic.com/steam/apps/1000030/header.jpg?t=1660866300
1000040,1000040,细胞战争,DoubleC Games,DoubleC Games,0,1,"0 .. 20,000",199,199,0,0,Simplified Chinese,"Action, Casual, Indie, Simulation",List(Single-player),"List(false, false, true)",https://cdn.akamai.steamstatic.com/steam/apps/1000040/header.jpg?t=1627033870
1000080,1000080,Zengeon,IndieLeague Studio,2P Games,1018,462,"100,000 .. 200,000",799,1999,60,3,"Simplified Chinese, English, Traditional Chinese, Japanese, Korean","Action, Adventure, Indie, RPG","List(Multi-player, Single-player, Steam Achievements, Full controller support, Steam Trading Cards)","List(false, true, true)",https://cdn.akamai.steamstatic.com/steam/apps/1000080/header.jpg?t=1667062553
1000100,1000100,干支セトラ 陽ノ卷｜干支etc. 陽之卷,七月九日,Starship Studio,18,6,"0 .. 20,000",1299,1299,0,0,"Japanese, Simplified Chinese, Traditional Chinese","Adventure, Indie, RPG, Strategy","List(Single-player, Steam Achievements, Steam Cloud)","List(false, false, true)",https://cdn.akamai.steamstatic.com/steam/apps/1000100/header.jpg?t=1561522270
1000110,1000110,Jumping Master(跳跳大咖),重庆环游者网络科技,重庆环游者网络科技,50,34,"20,000 .. 50,000",0,0,0,0,"English, Simplified Chinese, Traditional Chinese","Action, Adventure, Casual, Free to Play, Massively Multiplayer","List(Multi-player, Single-player, Co-op, Online PvP, Online Co-op, PvP)","List(false, false, true)",https://cdn.akamai.steamstatic.com/steam/apps/1000110/header.jpg?t=1562917106
1000130,1000130,Cube Defender,Simon Codrington,Simon Codrington,6,0,"0 .. 20,000",299,299,0,0,English,"Casual, Indie","List(Single-player, Steam Achievements, Steam Leaderboards)","List(false, true, true)",https://cdn.akamai.steamstatic.com/steam/apps/1000130/header.jpg?t=1646024900
1000280,1000280,Tower of Origin2-Worm's Nest,Villain Role,Villain Role,32,12,"0 .. 20,000",1399,1399,0,0,"English, Simplified Chinese, Traditional Chinese","Indie, RPG",List(Single-player),"List(false, false, true)",https://cdn.akamai.steamstatic.com/steam/apps/1000280/header.jpg?t=1649211613


In [0]:
# Statistiques de base
print("📊 STATISTIQUES DU DATASET")
print("=" * 60)

nb_jeux = df_clean.count()
nb_colonnes = len(df_clean.columns)

print(f"Nombre de jeux : {nb_jeux:,}")
print(f"Nombre de colonnes : {nb_colonnes}")
print("=" * 60)

# Liste des colonnes
print("\n📋 COLONNES DISPONIBLES :")
for i, col_name in enumerate(df_clean.columns, 1):
    print(f"  {i}. {col_name}")

📊 STATISTIQUES DU DATASET
Nombre de jeux : 55,691
Nombre de colonnes : 17

📋 COLONNES DISPONIBLES :
  1. id
  2. appid
  3. name
  4. developer
  5. publisher
  6. positive
  7. negative
  8. owners
  9. price
  10. initialprice
  11. discount
  12. ccu
  13. languages
  14. genre
  15. categories
  16. platforms
  17. header_image


In [0]:
# Compter les valeurs NULL pour chaque colonne
print("\n🕳️ VALEURS MANQUANTES PAR COLONNE")
print("=" * 60)

missing_counts = df_clean.select([
    count(when(col(c).isNull(), c)).alias(c) 
    for c in df_clean.columns
])

# Afficher sous forme de tableau
display(missing_counts)


🕳️ VALEURS MANQUANTES PAR COLONNE


id,appid,name,developer,publisher,positive,negative,owners,price,initialprice,discount,ccu,languages,genre,categories,platforms,header_image
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0


In [0]:
# Regarder quelques noms de jeux
print("🎮 EXEMPLES DE NOMS DE JEUX :")
print("=" * 60)

df_clean.select("name").show(20, truncate=False)

# Compter les jeux uniques
nb_jeux_uniques = df_clean.select("name").distinct().count()
print(f"\n📊 Nombre de jeux uniques : {nb_jeux_uniques:,}")

🎮 EXEMPLES DE NOMS DE JEUX :
+----------------------------------------------+
|name                                          |
+----------------------------------------------+
|Counter-Strike                                |
|ASCENXION                                     |
|Crown Trick                                   |
|Cook, Serve, Delicious! 3?!                   |
|细胞战争                                      |
|Zengeon                                       |
|干支セトラ　陽ノ卷｜干支etc.　陽之卷          |
|Jumping Master(跳跳大咖)                      |
|Cube Defender                                 |
|Tower of Origin2-Worm's Nest                  |
|人气动漫大乱斗                                |
|Hellish Quart                                 |
|SurReal Subway                                |
|Rogue Reaper                                  |
|WRATH: Aeon of Ruin                           |
|Drawngeon: Dungeons of Ink and Paper          |
|Battle Motion                                 |
|The Undisputables : O

In [0]:
# Statistiques sur les prix
print("💰 STATISTIQUES DES PRIX :")
print("=" * 60)

df_clean.select(
    count("price").alias("Nb_jeux_avec_prix"),
    round(avg("price"), 2).alias("Prix_moyen"),
    min("price").alias("Prix_min"),
    max("price").alias("Prix_max")
).show()

# Voir quelques exemples de prix
df_clean.select("name", "price", "initialprice", "discount") \
    .filter(col("price").isNotNull()) \
    .show(10)

💰 STATISTIQUES DES PRIX :
+-----------------+----------+--------+--------+
|Nb_jeux_avec_prix|Prix_moyen|Prix_min|Prix_max|
+-----------------+----------+--------+--------+
|            55691|    773.28|       0|    9999|
+-----------------+----------+--------+--------+

+------------------------------------+-----+------------+--------+
|                                name|price|initialprice|discount|
+------------------------------------+-----+------------+--------+
|                      Counter-Strike|  999|         999|       0|
|                           ASCENXION|  999|         999|       0|
|                         Crown Trick|  599|        1999|      70|
|                Cook, Serve, Deli...| 1999|        1999|       0|
|                            细胞战争|  199|         199|       0|
|                             Zengeon|  799|        1999|      60|
|干支セトラ　陽ノ卷｜干支etc.　陽之卷| 1299|        1299|       0|
|            Jumping Master(跳跳大咖)|    0|           0|       0|
|              

In [0]:
# ========================================
# ⭐ ANALYSE DES AVIS (CORRIGÉ)
# ========================================

# Créer un ratio positif en gérant les divisions par zéro
df_reviews = df_clean.withColumn(
    "total_reviews",
    col("positive") + col("negative")
).withColumn(
    "ratio_positif",
    when(
        (col("positive") + col("negative")) == 0, 
        None
    ).otherwise(
        round(col("positive") / (col("positive") + col("negative")) * 100, 2)
    )
)

# Statistiques générales
print("⭐ STATISTIQUES DES AVIS :")
print("=" * 60)

df_reviews.select(
    count("positive").alias("Jeux_avec_avis"),
    round(avg("positive"), 2).alias("Avis_positifs_moyen"),
    round(avg("negative"), 2).alias("Avis_negatifs_moyen"),
    round(avg("ratio_positif"), 2).alias("Ratio_positif_moyen")
).show()

# Top 10 jeux les mieux notés (avec au moins 100 avis)
print("\n🏆 TOP 10 JEUX LES MIEUX NOTÉS :")
print("=" * 60)

df_reviews.select("name", "positive", "negative", "total_reviews", "ratio_positif") \
    .filter(col("total_reviews") > 100) \
    .filter(col("ratio_positif").isNotNull()) \
    .orderBy(desc("ratio_positif")) \
    .show(10, truncate=False)

⭐ STATISTIQUES DES AVIS :
+--------------+-------------------+-------------------+-------------------+
|Jeux_avec_avis|Avis_positifs_moyen|Avis_negatifs_moyen|Ratio_positif_moyen|
+--------------+-------------------+-------------------+-------------------+
|         55691|            1470.88|             241.84|              73.65|
+--------------+-------------------+-------------------+-------------------+


🏆 TOP 10 JEUX LES MIEUX NOTÉS :
+-------------------------------------------------+--------+--------+-------------+-------------+
|name                                             |positive|negative|total_reviews|ratio_positif|
+-------------------------------------------------+--------+--------+-------------+-------------+
|Cyberfrags '69                                   |104     |0       |104          |100.0        |
|Elasto Mania Remastered                          |190     |0       |190          |100.0        |
|Lucy Dreaming                                    |118     |0    

In [0]:
# Voir la structure des genres
print("🎭 EXEMPLES DE GENRES :")
print("=" * 60)

df_clean.select("name", "genre").show(10, truncate=False)

# Compter combien de jeux ont des genres
jeux_avec_genres = df_clean.filter(col("genre").isNotNull()).count()
print(f"\n📊 Jeux avec genres renseignés : {jeux_avec_genres:,}")

🎭 EXEMPLES DE GENRES :
+------------------------------------+--------------------------------------------------------------+
|name                                |genre                                                         |
+------------------------------------+--------------------------------------------------------------+
|Counter-Strike                      |Action                                                        |
|ASCENXION                           |Action, Adventure, Indie                                      |
|Crown Trick                         |Adventure, Indie, RPG, Strategy                               |
|Cook, Serve, Delicious! 3?!         |Action, Indie, Simulation, Strategy                           |
|细胞战争                            |Action, Casual, Indie, Simulation                             |
|Zengeon                             |Action, Adventure, Indie, RPG                                 |
|干支セトラ　陽ノ卷｜干支etc.　陽之卷|Adventure, Indie, RPG, Strategy          

In [0]:
# Voir la structure des plateformes
print("💻 EXEMPLES DE PLATEFORMES :")
print("=" * 60)

df_clean.select("platforms").printSchema()
df_clean.select("name", "platforms").show(5, truncate=False)

# Extraire les infos de plateformes
df_platforms = df_clean.select(
    "name",
    col("platforms.windows").alias("windows"),
    col("platforms.mac").alias("mac"),
    col("platforms.linux").alias("linux")
)

# Compter par plateforme
df_platforms.select(
    sum(when(col("windows") == True, 1).otherwise(0)).alias("Jeux_Windows"),
    sum(when(col("mac") == True, 1).otherwise(0)).alias("Jeux_Mac"),
    sum(when(col("linux") == True, 1).otherwise(0)).alias("Jeux_Linux")
).show()

💻 EXEMPLES DE PLATEFORMES :
root
 |-- platforms: struct (nullable = true)
 |    |-- linux: boolean (nullable = true)
 |    |-- mac: boolean (nullable = true)
 |    |-- windows: boolean (nullable = true)

+---------------------------+--------------------+
|name                       |platforms           |
+---------------------------+--------------------+
|Counter-Strike             |{true, true, true}  |
|ASCENXION                  |{false, false, true}|
|Crown Trick                |{false, false, true}|
|Cook, Serve, Delicious! 3?!|{false, true, true} |
|细胞战争                   |{false, false, true}|
+---------------------------+--------------------+
only showing top 5 rows
+------------+--------+----------+
|Jeux_Windows|Jeux_Mac|Jeux_Linux|
+------------+--------+----------+
|       55676|   12770|      8458|
+------------+--------+----------+



In [0]:
# Voir la structure des catégories
print("📂 EXEMPLES DE CATÉGORIES :")
print("=" * 60)

df_clean.select("categories").printSchema()
df_clean.select("name", "categories").show(5, truncate=False)

📂 EXEMPLES DE CATÉGORIES :
root
 |-- categories: array (nullable = true)
 |    |-- element: string (containsNull = true)

+---------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|name                       |categories                                                                                                                                                                                                                                               |
+---------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|Counter-Strike             |[Multi-pl

In [0]:
# Résumé final
print("📊 RÉSUMÉ DE LA QUALITÉ DES DONNÉES")
print("=" * 60)

quality_summary = df_clean.select(
    count("name").alias("Total_jeux"),
    count("developer").alias("Avec_developer"),
    count("publisher").alias("Avec_publisher"),
    count("price").alias("Avec_price"),
    count("genre").alias("Avec_genre"),
    count("platforms").alias("Avec_platforms"),
    count("positive").alias("Avec_avis")
)

display(quality_summary)

print("\n✅ Exploration terminée !")
print("💡 Dataframe 'df_clean' prêt pour les analyses !")

📊 RÉSUMÉ DE LA QUALITÉ DES DONNÉES


Total_jeux,Avec_developer,Avec_publisher,Avec_price,Avec_genre,Avec_platforms,Avec_avis
55691,55691,55691,55691,55691,55691,55691



✅ Exploration terminée !
💡 Dataframe 'df_clean' prêt pour les analyses !


In [0]:
# ========================================
# 🏢 ANALYSE DES ÉDITEURS (PUBLISHERS)
# ========================================

print("🏢 TOP 20 ÉDITEURS AVEC LE PLUS DE JEUX")
print("=" * 60)

# Compter les jeux par éditeur
top_publishers = df_clean.groupBy("publisher") \
    .agg(count("*").alias("nombre_jeux")) \
    .filter(col("publisher").isNotNull()) \
    .orderBy(desc("nombre_jeux")) \
    .limit(20)

display(top_publishers)

🏢 TOP 20 ÉDITEURS AVEC LE PLUS DE JEUX


publisher,nombre_jeux
Big Fish Games,422
8floor,202
SEGA,165
Strategy First,151
Square Enix,141
Choice of Games,140
Sekai Project,132
,132
HH-Games,132
Ubisoft,127


Databricks visualization. Run in Databricks to view.

In [0]:
# ========================================
# ⭐ TOP JEUX LES MIEUX NOTÉS
# ========================================

print("🏆 TOP 20 JEUX LES MIEUX NOTÉS (avec au moins 500 avis)")
print("=" * 60)

# On réutilise df_reviews créé à l'étape 2
best_rated_games = df_reviews.select(
    "name", 
    "positive", 
    "negative", 
    "total_reviews", 
    "ratio_positif"
) \
    .filter(col("total_reviews") >= 500) \
    .filter(col("ratio_positif").isNotNull()) \
    .orderBy(desc("ratio_positif")) \
    .limit(20)

display(best_rated_games)

🏆 TOP 20 JEUX LES MIEUX NOTÉS (avec au moins 500 avis)


name,positive,negative,total_reviews,ratio_positif
Flowers -Le volume sur ete-,937,1,938,99.89
Touhou Fuujinroku ~ Mountain of Faith.,649,4,653,99.39
Aventura Copilului Albastru și Urât,2203,14,2217,99.37
The Case of the Golden Idol,909,6,915,99.34
Aseprite,11823,80,11903,99.33
Patrick's Parabox,1489,11,1500,99.27
A Short Hike,11645,87,11732,99.26
Monster Prom 3: Monster Roadtrip,1052,8,1060,99.25
Flowers -Le volume sur automne-,516,4,520,99.23
CULTIC,2021,16,2037,99.21


Databricks visualization. Run in Databricks to view.

In [0]:
# ========================================
# 💰 ANALYSE DES PRIX
# ========================================

print("💰 STATISTIQUES DES PRIX")
print("=" * 60)

# Statistiques générales sur les prix
price_stats = df_clean.select(
    count("price").alias("Jeux_avec_prix"),
    round(avg("price"), 2).alias("Prix_moyen"),
    round(min("price"), 2).alias("Prix_min"),
    round(max("price"), 2).alias("Prix_max")
).show()

# Distribution des prix par tranches
print("\n💵 RÉPARTITION DES JEUX PAR TRANCHE DE PRIX")
print("=" * 60)

price_distribution = df_clean.withColumn(
    "tranche_prix",
    when(col("price") == 0, "Gratuit")
    .when((col("price") > 0) & (col("price") <= 5), "0-5€")
    .when((col("price") > 5) & (col("price") <= 10), "5-10€")
    .when((col("price") > 10) & (col("price") <= 20), "10-20€")
    .when((col("price") > 20) & (col("price") <= 40), "20-40€")
    .when(col("price") > 40, "+40€")
    .otherwise("Non renseigné")
) \
    .groupBy("tranche_prix") \
    .agg(count("*").alias("nombre_jeux")) \
    .orderBy("nombre_jeux", ascending=False)

display(price_distribution)

💰 STATISTIQUES DES PRIX
+--------------+----------+--------+--------+
|Jeux_avec_prix|Prix_moyen|Prix_min|Prix_max|
+--------------+----------+--------+--------+
|         55691|    773.28|     0.0|  9999.0|
+--------------+----------+--------+--------+


💵 RÉPARTITION DES JEUX PAR TRANCHE DE PRIX


tranche_prix,nombre_jeux
+40€,47839
Gratuit,7780
20-40€,72


Databricks visualization. Run in Databricks to view.

Databricks visualization. Run in Databricks to view.

In [0]:
# ========================================
# 🎁 ANALYSE DES PROMOTIONS
# ========================================

print("🎁 JEUX AVEC ET SANS PROMOTIONS")
print("=" * 60)

promo_analysis = df_clean.withColumn(
    "statut_promo",
    when(col("discount") > 0, "Avec promotion")
    .when(col("discount") == 0, "Sans promotion")
    .otherwise("Non renseigné")
) \
    .groupBy("statut_promo") \
    .agg(count("*").alias("nombre_jeux"))

display(promo_analysis)

# Statistiques sur les réductions
print("\n💸 STATISTIQUES SUR LES RÉDUCTIONS")
print("=" * 60)

discount_stats = df_clean.filter(col("discount") > 0).select(
    count("discount").alias("Jeux_en_promo"),
    round(avg("discount"), 2).alias("Réduction_moyenne_%"),
    round(min("discount"), 2).alias("Réduction_min_%"),
    round(max("discount"), 2).alias("Réduction_max_%")
)

display(discount_stats)

🎁 JEUX AVEC ET SANS PROMOTIONS


statut_promo,nombre_jeux
Sans promotion,53173
Avec promotion,2518


Databricks visualization. Run in Databricks to view.


💸 STATISTIQUES SUR LES RÉDUCTIONS


Jeux_en_promo,Réduction_moyenne_%,Réduction_min_%,Réduction_max_%
2518,57.59,10.0,90.0


In [0]:
# ========================================
# 🌍 ANALYSE DES LANGUES
# ========================================

print("🌍 LANGUES LES PLUS REPRÉSENTÉES")
print("=" * 60)

# D'abord, voir la structure de la colonne languages
df_clean.select("languages").show(5, truncate=False)

# Si languages est un STRING avec des langues séparées par des virgules :
# On va split et explode
df_languages = df_clean.select(
    "name",
    explode(split(col("languages"), ",")).alias("langue")
) \
    .withColumn("langue", trim(col("langue")))  # Enlever les espaces

# Compter par langue
top_languages = df_languages.groupBy("langue") \
    .agg(count("*").alias("nombre_jeux")) \
    .filter(col("langue") != "") \
    .orderBy(desc("nombre_jeux")) \
    .limit(15)

display(top_languages)

🌍 LANGUES LES PLUS REPRÉSENTÉES
+-------------------------------------------------------------------------------------------------------------------------+
|languages                                                                                                                |
+-------------------------------------------------------------------------------------------------------------------------+
|English, French, German, Italian, Spanish - Spain, Simplified Chinese, Traditional Chinese, Korean                       |
|English, Korean, Simplified Chinese                                                                                      |
|Simplified Chinese, English, Japanese, Traditional Chinese, French, German, Spanish - Spain, Russian, Portuguese - Brazil|
|English                                                                                                                  |
|Simplified Chinese                                                                                 

langue,nombre_jeux
English,55116
German,14019
French,13426
Russian,12922
Simplified Chinese,12782
Spanish - Spain,12233
Japanese,10368
Italian,9304
Portuguese - Brazil,6750
Korean,6600


Databricks visualization. Run in Databricks to view.

In [0]:
# ========================================
# 🔞 ANALYSE DES CATÉGORIES (SOLUTION FINALE)
# ========================================

from pyspark.sql.functions import explode, lower, col, count, desc

print("🔞 ANALYSE DES CATÉGORIES")
print("=" * 60)

# Exploser les catégories
df_categories = df_clean.select(
    "name",
    explode(col("categories")).alias("categorie")
)

# Top 20 catégories
print("📊 TOP 20 CATÉGORIES :")
top_categories = df_categories.groupBy("categorie") \
    .agg(count("*").alias("nombre_jeux")) \
    .orderBy(desc("nombre_jeux")) \
    .limit(20)

display(top_categories)

# Chercher les catégories "sensibles"
print("\n🔍 RECHERCHE DE CONTENU MATURE :")

mature_categories = df_categories.filter(
    lower(col("categorie")).contains("mature") |
    lower(col("categorie")).contains("adult") |
    lower(col("categorie")).contains("violent")
)

mature_count = mature_categories.select("name").distinct().count()
total_games = df_clean.count()

print(f"Jeux avec catégories sensibles : {mature_count}")
print(f"Total jeux : {total_games}")

# Calcul ULTRA sécurisé du pourcentage
try:
    if total_games > 0:
        percentage = round((mature_count / total_games) * 100, 2)
        print(f"Pourcentage : {percentage}%")
    else:
        print("Pourcentage : N/A")
except Exception as e:
    print(f"Pourcentage : 0%")

🔞 ANALYSE DES CATÉGORIES
📊 TOP 20 CATÉGORIES :


categorie,nombre_jeux
Single-player,52025
Steam Achievements,27394
Steam Cloud,14235
Full controller support,11879
Multi-player,11455
Steam Trading Cards,9208
Partial Controller Support,7867
PvP,7070
Co-op,5616
Steam Leaderboards,5509


Databricks visualization. Run in Databricks to view.


🔍 RECHERCHE DE CONTENU MATURE :
Jeux avec catégories sensibles : 0
Total jeux : 55691
Pourcentage : 0%


In [0]:
# ========================================
# 🎯 RÉSUMÉ GÉNÉRAL DU MARCHÉ STEAM
# ========================================

print("🎯 SYNTHÈSE GÉNÉRALE")
print("=" * 60)

summary = df_clean.select(
    count("name").alias("Total_jeux"),
    countDistinct("publisher").alias("Nombre_éditeurs"),
    round(avg("price"), 2).alias("Prix_moyen"),
    sum(when(col("discount") > 0, 1).otherwise(0)).alias("Jeux_en_promo"),
    round(avg("positive"), 0).alias("Avis_positifs_moyen"),
    round(avg("negative"), 0).alias("Avis_négatifs_moyen")
)

display(summary)

print("\n✅ Analyses macro terminées !")

🎯 SYNTHÈSE GÉNÉRALE


Total_jeux,Nombre_éditeurs,Prix_moyen,Jeux_en_promo,Avis_positifs_moyen,Avis_négatifs_moyen
55691,29966,773.28,2518,1471.0,242.0



✅ Analyses macro terminées !


In [0]:
# ========================================
# 🎮 ANALYSE DES GENRES
# ========================================

from pyspark.sql.functions import explode, col, count, desc, avg, sum, round

print("🎮 ANALYSE DES GENRES - PARTIE 1")
print("=" * 60)

# D'abord, voir la structure de la colonne genre
print("🔍 Structure de la colonne 'genre' :")
df_clean.select("genre").show(5, truncate=False)

# Vérifier le type
df_clean.select("genre").printSchema()

🎮 ANALYSE DES GENRES - PARTIE 1
🔍 Structure de la colonne 'genre' :
+-----------------------------------+
|genre                              |
+-----------------------------------+
|Action                             |
|Action, Adventure, Indie           |
|Adventure, Indie, RPG, Strategy    |
|Action, Indie, Simulation, Strategy|
|Action, Casual, Indie, Simulation  |
+-----------------------------------+
only showing top 5 rows
root
 |-- genre: string (nullable = true)



In [0]:
# Si genre = "Action, Adventure, RPG" (texte)
df_genres = df_clean.select(
    "name",
    "price",
    "positive",
    "negative",
    explode(split(col("genre"), ",")).alias("genre_simple")
) \
    .withColumn("genre_simple", trim(col("genre_simple")))  # Enlever les espaces

# Top 20 genres
print("🏆 TOP 20 GENRES LES PLUS REPRÉSENTÉS :")
print("=" * 60)

top_genres = df_genres.groupBy("genre_simple") \
    .agg(count("*").alias("nombre_jeux")) \
    .filter(col("genre_simple") != "") \
    .orderBy(desc("nombre_jeux")) \
    .limit(20)

display(top_genres)

🏆 TOP 20 GENRES LES PLUS REPRÉSENTÉS :


genre_simple,nombre_jeux
Indie,39681
Action,23759
Casual,22086
Adventure,21431
Strategy,10895
Simulation,10836
RPG,9534
Early Access,6145
Free to Play,3393
Sports,2666


Databricks visualization. Run in Databricks to view.

In [0]:
# ========================================
# ⭐ GENRES AVEC MEILLEURS AVIS
# ========================================

print("⭐ GENRES LES MIEUX NOTÉS (avec au moins 50 jeux)")
print("=" * 60)

# Filtrer les jeux avec des avis valides
genres_ratings = df_genres.filter(
    (col("positive").isNotNull()) & 
    (col("negative").isNotNull()) &
    (col("positive") + col("negative") > 0)  # IMPORTANT : éviter division par zéro
) \
    .withColumn("total_avis", col("positive") + col("negative")) \
    .withColumn(
        "ratio_positif",
        round((col("positive") / col("total_avis")) * 100, 2)
    )

# Grouper par genre et calculer les moyennes
best_genres = genres_ratings.groupBy("genre_simple") \
    .agg(
        count("*").alias("nombre_jeux"),
        round(avg("positive"), 0).alias("avis_positifs_moyen"),
        round(avg("negative"), 0).alias("avis_negatifs_moyen"),
        round(avg("ratio_positif"), 2).alias("ratio_positif_moyen")
    ) \
    .filter(col("nombre_jeux") >= 50) \
    .orderBy(desc("ratio_positif_moyen")) \
    .limit(20)

display(best_genres)

⭐ GENRES LES MIEUX NOTÉS (avec au moins 50 jeux)


genre_simple,nombre_jeux,avis_positifs_moyen,avis_negatifs_moyen,ratio_positif_moyen
Casual,21990,456.0,70.0,74.34
Indie,39558,822.0,107.0,74.21
Adventure,21397,1388.0,264.0,73.88
Game Development,159,173.0,21.0,73.78
RPG,9520,2040.0,344.0,73.06
Action,23695,2315.0,409.0,73.0
Free to Play,3388,5526.0,1264.0,72.19
Design & Illustration,405,1664.0,67.0,72.07
Web Publishing,89,382.0,59.0,71.97
Strategy,10876,1232.0,220.0,71.93


Databricks visualization. Run in Databricks to view.

In [0]:
# ========================================
# 💰 GENRES LES PLUS CHERS
# ========================================

print("💰 PRIX MOYEN PAR GENRE (top 20)")
print("=" * 60)

# Prix moyen par genre
price_by_genre = df_genres.filter(
    (col("price").isNotNull()) & 
    (col("price") > 0)  # Exclure les jeux gratuits si tu veux
) \
    .groupBy("genre_simple") \
    .agg(
        count("*").alias("nombre_jeux"),
        round(avg("price"), 2).alias("prix_moyen"),
        round(min("price"), 2).alias("prix_min"),
        round(max("price"), 2).alias("prix_max")
    ) \
    .filter(col("nombre_jeux") >= 30) \
    .orderBy(desc("prix_moyen")) \
    .limit(20)

display(price_by_genre)

print("\n💸 GENRES LES MOINS CHERS :")
cheapest_genres = df_genres.filter(
    (col("price").isNotNull()) & 
    (col("price") > 0)
) \
    .groupBy("genre_simple") \
    .agg(
        count("*").alias("nombre_jeux"),
        round(avg("price"), 2).alias("prix_moyen")
    ) \
    .filter(col("nombre_jeux") >= 30) \
    .orderBy("prix_moyen") \
    .limit(10)

display(cheapest_genres)

💰 PRIX MOYEN PAR GENRE (top 20)


genre_simple,nombre_jeux,prix_moyen,prix_min,prix_max
Video Production,133,3520.94,1099.0,9999.0
Audio Production,122,3139.98,12499.0,9999.0
Animation & Modeling,196,3084.68,1099.0,9999.0
Web Publishing,64,3031.05,1199.0,9999.0
Photo Editing,74,2884.12,1099.0,9999.0
Game Development,119,2843.59,1099.0,9999.0
Design & Illustration,286,2705.49,1099.0,9999.0
Software Training,117,2667.83,1099.0,9999.0
Education,190,2392.8,1099.0,9999.0
Utilities,461,1693.97,1099.0,9999.0


Databricks visualization. Run in Databricks to view.


💸 GENRES LES MOINS CHERS :


genre_simple,nombre_jeux,prix_moyen
Violent,168,527.96
Gore,99,570.06
Casual,19418,637.74
Free to Play,149,652.57
Sexual Content,54,687.52
Nudity,45,695.89
Indie,34765,749.69
,83,754.43
Action,20582,891.98
Adventure,19019,902.16


Databricks visualization. Run in Databricks to view.

In [0]:
# ========================================
# 🎯 ESTIMATION DES REVENUS PAR GENRE
# ========================================

print("🎯 GENRES LES PLUS LUCRATIFS (estimation)")
print("=" * 60)

# Estimer le revenu : prix × nombre d'avis (proxy des ventes)
revenue_by_genre = df_genres.filter(
    (col("price").isNotNull()) & 
    (col("positive").isNotNull()) & 
    (col("negative").isNotNull()) &
    (col("price") > 0) &
    ((col("positive") + col("negative")) > 0)  # Éviter divisions par zéro
) \
    .withColumn("total_avis", col("positive") + col("negative")) \
    .withColumn("revenu_estime", col("price") * col("total_avis")) \
    .groupBy("genre_simple") \
    .agg(
        count("*").alias("nombre_jeux"),
        round(sum("revenu_estime"), 0).alias("revenu_total_estime"),
        round(avg("revenu_estime"), 0).alias("revenu_moyen_par_jeu"),
        round(avg("price"), 2).alias("prix_moyen")
    ) \
    .filter(col("nombre_jeux") >= 30) \
    .orderBy(desc("revenu_total_estime")) \
    .limit(15)

display(revenue_by_genre)

🎯 GENRES LES PLUS LUCRATIFS (estimation)


genre_simple,nombre_jeux,revenu_total_estime,revenu_moyen_par_jeu,prix_moyen
Action,20523,105306097551,5131126.0,892.92
Adventure,18986,67822663580,3572246.0,902.28
Indie,34649,50648939283,1461772.0,750.12
RPG,8127,49239517700,6058757.0,1059.1
Simulation,9427,34066895559,3613758.0,1043.11
Strategy,9367,30288584895,3233542.0,975.85
Massively Multiplayer,686,11609725801,1.6923799E7,1064.19
Casual,19332,10335681222,534641.0,638.2
Early Access,5032,8751852760,1739239.0,1065.74
Sports,2248,5755989868,2560494.0,1060.02


Databricks visualization. Run in Databricks to view.

In [0]:
# ========================================
# 📊 COMPARAISON : Action vs RPG vs Indie vs Strategy (CORRIGÉ)
# ========================================

print("📊 COMPARAISON : Action vs RPG vs Indie vs Strategy")
print("=" * 60)

# Filtrer sur les genres les plus populaires
popular_genres = ["Action", "RPG", "Indie", "Strategy", "Adventure", "Simulation"]

genre_comparison = df_genres.filter(
    col("genre_simple").isin(popular_genres)
) \
    .filter(
        (col("positive").isNotNull()) & 
        (col("negative").isNotNull()) &
        (col("price").isNotNull()) &
        ((col("positive") + col("negative")) > 0)  # Protection division par zéro
    ) \
    .withColumn("total_avis", col("positive") + col("negative")) \
    .withColumn("ratio_positif", (col("positive") / col("total_avis")) * 100) \
    .groupBy("genre_simple") \
    .agg(
        count("*").alias("nombre_jeux"),
        round(avg("price"), 2).alias("prix_moyen"),
        round(avg("ratio_positif"), 2).alias("ratio_positif_moyen"),
        round(avg("total_avis"), 0).alias("avis_moyen")
    ) \
    .orderBy(desc("nombre_jeux"))

display(genre_comparison)

📊 COMPARAISON : Action vs RPG vs Indie vs Strategy


genre_simple,nombre_jeux,prix_moyen,ratio_positif_moyen,avis_moyen
Indie,39558,657.03,74.21,930.0
Action,23695,773.38,73.0,2724.0
Adventure,21397,800.61,73.88,1652.0
Strategy,10876,840.45,71.93,1452.0
Simulation,10809,909.74,69.37,1663.0
RPG,9520,904.13,73.06,2384.0


In [0]:
# ========================================
# 🏢 GENRES PRÉFÉRÉS DES TOP ÉDITEURS (CORRIGÉ)
# ========================================

print("🏢 GENRES PRÉFÉRÉS DES TOP ÉDITEURS")
print("=" * 60)

# IMPORTANT : Recréer df_genres AVEC publisher cette fois
df_genres_with_publisher = df_clean.select(
    "name",
    "publisher",
    explode(split(col("genre"), ",")).alias("genre_simple")
) \
    .withColumn("genre_simple", trim(col("genre_simple"))) \
    .filter(
        (col("publisher").isNotNull()) &
        (col("genre_simple").isNotNull()) &
        (col("genre_simple") != "")
    )

# Top 10 éditeurs
top_10_publishers = df_clean.groupBy("publisher") \
    .agg(count("*").alias("nb_jeux")) \
    .filter(col("publisher").isNotNull()) \
    .orderBy(desc("nb_jeux")) \
    .limit(10)

# Extraire juste les noms des publishers
top_publishers_list = [row.publisher for row in top_10_publishers.collect()]

# Filtrer sur les top 10 publishers
publisher_genre_analysis = df_genres_with_publisher.filter(
    col("publisher").isin(top_publishers_list)
) \
    .groupBy("publisher", "genre_simple") \
    .agg(count("*").alias("nombre_jeux")) \
    .orderBy("publisher", desc("nombre_jeux"))

display(publisher_genre_analysis)

🏢 GENRES PRÉFÉRÉS DES TOP ÉDITEURS


publisher,genre_simple,nombre_jeux
,Indie,106
,Action,70
,Casual,64
,Adventure,37
,Simulation,23
,Strategy,19
,RPG,16
,Early Access,11
,Racing,8
,Sports,6


In [0]:
# ========================================
# 💻 ANALYSE DES PLATEFORMES
# ========================================

print("💻 ANALYSE DES PLATEFORMES")
print("=" * 60)

# Voir la structure de la colonne platforms
print("🔍 Structure de la colonne 'platforms' :")
df_clean.select("platforms").printSchema()

# Voir quelques exemples
print("\n👀 Exemples de plateformes :")
df_clean.select("name", "platforms").show(5, truncate=False)

💻 ANALYSE DES PLATEFORMES
🔍 Structure de la colonne 'platforms' :
root
 |-- platforms: struct (nullable = true)
 |    |-- linux: boolean (nullable = true)
 |    |-- mac: boolean (nullable = true)
 |    |-- windows: boolean (nullable = true)


👀 Exemples de plateformes :
+---------------------------+--------------------+
|name                       |platforms           |
+---------------------------+--------------------+
|Counter-Strike             |{true, true, true}  |
|ASCENXION                  |{false, false, true}|
|Crown Trick                |{false, false, true}|
|Cook, Serve, Delicious! 3?!|{false, true, true} |
|细胞战争                   |{false, false, true}|
+---------------------------+--------------------+
only showing top 5 rows


In [0]:
# ========================================
# 📊 ANALYSE COMPLÈTE DES PLATEFORMES
# ========================================

print("📊 NOMBRE DE JEUX PAR PLATEFORME")
print("=" * 60)

# 1. Extraire les booléens des plateformes
df_platforms = df_clean.select(
    "name",
    col("platforms.windows").alias("windows"),
    col("platforms.mac").alias("mac"),
    col("platforms.linux").alias("linux")
)

# 2. Afficher quelques exemples
print("\n👀 Exemples de plateformes :")
df_platforms.show(5)

# 3. CALCULER LES NOMBRES (AVEC CONVERSION INT)
jeux_windows = int(df_platforms.filter(col("windows") == True).count())
jeux_mac = int(df_platforms.filter(col("mac") == True).count())
jeux_linux = int(df_platforms.filter(col("linux") == True).count())
total_games = int(df_clean.count())

# 4. MAINTENANT on peut afficher les statistiques
print("\n📊 STATISTIQUES :")
print("=" * 60)
print(f"Total de jeux : {total_games}")

print("\nRépartition par plateforme :")
print(f"Windows : {jeux_windows} jeux ({__builtins__.round(jeux_windows / total_games * 100, 2)}%)")
print(f"Mac : {jeux_mac} jeux ({__builtins__.round(jeux_mac / total_games * 100, 2)}%)")
print(f"Linux : {jeux_linux} jeux ({__builtins__.round(jeux_linux / total_games * 100, 2)}%)")

# 5. Créer un dataframe pour la visualisation
platform_data = [
    Row(Plateforme="Windows", Nombre_jeux=jeux_windows, Pourcentage=__builtins__.round(jeux_windows / total_games * 100, 1)),
    Row(Plateforme="Mac", Nombre_jeux=jeux_mac, Pourcentage=__builtins__.round(jeux_mac / total_games * 100, 1)),
    Row(Plateforme="Linux", Nombre_jeux=jeux_linux, Pourcentage=__builtins__.round(jeux_linux / total_games * 100, 1))
]

df_platform_viz = spark.createDataFrame(platform_data)

print("\n📊 TABLEAU POUR VISUALISATION :")
display(df_platform_viz)

📊 NOMBRE DE JEUX PAR PLATEFORME

👀 Exemples de plateformes :
+--------------------+-------+-----+-----+
|                name|windows|  mac|linux|
+--------------------+-------+-----+-----+
|      Counter-Strike|   true| true| true|
|           ASCENXION|   true|false|false|
|         Crown Trick|   true|false|false|
|Cook, Serve, Deli...|   true| true|false|
|            细胞战争|   true|false|false|
+--------------------+-------+-----+-----+
only showing top 5 rows

📊 STATISTIQUES :
Total de jeux : 55691

Répartition par plateforme :
Windows : 55676 jeux (99.97%)
Mac : 12770 jeux (22.93%)
Linux : 8458 jeux (15.19%)

📊 TABLEAU POUR VISUALISATION :


Plateforme,Nombre_jeux,Pourcentage
Windows,55676,100.0
Mac,12770,22.9
Linux,8458,15.2


Databricks visualization. Run in Databricks to view.

In [0]:
# ========================================
# 🎮 ANALYSE MULTI-PLATEFORMES
# ========================================

print("🎮 JEUX MULTI-PLATEFORMES VS EXCLUSIFS")
print("=" * 60)

# Ajouter une colonne pour compter les plateformes
df_platform_analysis = df_platforms.withColumn(
    "nb_platforms",
    when(col("windows") == True, 1).otherwise(0) +
    when(col("mac") == True, 1).otherwise(0) +
    when(col("linux") == True, 1).otherwise(0)
) \
    .withColumn(
        "type_plateforme",
        when(col("nb_platforms") == 3, "Toutes plateformes (Win+Mac+Linux)")
        .when(col("nb_platforms") == 2, "2 plateformes")
        .when(col("nb_platforms") == 1, "Exclusif 1 plateforme")
        .otherwise("Aucune plateforme")
    )

# Compter par type
platform_distribution = df_platform_analysis.groupBy("type_plateforme") \
    .agg(count("*").alias("nombre_jeux")) \
    .orderBy(desc("nombre_jeux"))

display(platform_distribution)

🎮 JEUX MULTI-PLATEFORMES VS EXCLUSIFS


type_plateforme,nombre_jeux
Exclusif 1 plateforme,41285
2 plateformes,7599
Toutes plateformes (Win+Mac+Linux),6807


Databricks visualization. Run in Databricks to view.

In [0]:
# ========================================
# 💰 PRIX MOYEN PAR PLATEFORME
# ========================================

print("💰 PRIX MOYEN PAR PLATEFORME")
print("=" * 60)

# Joindre avec les prix
df_platform_prices = df_clean.select(
    "name",
    "price",
    col("platforms.windows").alias("windows"),
    col("platforms.mac").alias("mac"),
    col("platforms.linux").alias("linux")
).filter(col("price").isNotNull())

# Prix moyen pour les jeux Windows
windows_price = df_platform_prices.filter(col("windows") == True) \
    .select(round(avg("price"), 2).alias("Prix_moyen_Windows"))

# Prix moyen pour les jeux Mac
mac_price = df_platform_prices.filter(col("mac") == True) \
    .select(round(avg("price"), 2).alias("Prix_moyen_Mac"))

# Prix moyen pour les jeux Linux
linux_price = df_platform_prices.filter(col("linux") == True) \
    .select(round(avg("price"), 2).alias("Prix_moyen_Linux"))

print("💵 COMPARAISON DES PRIX MOYENS :")
windows_price.show()
mac_price.show()
linux_price.show()

💰 PRIX MOYEN PAR PLATEFORME
💵 COMPARAISON DES PRIX MOYENS :
+------------------+
|Prix_moyen_Windows|
+------------------+
|            772.91|
+------------------+

+--------------+
|Prix_moyen_Mac|
+--------------+
|        778.95|
+--------------+

+----------------+
|Prix_moyen_Linux|
+----------------+
|           779.4|
+----------------+



In [0]:
# ========================================
# ⭐ QUALITÉ DES JEUX PAR PLATEFORME
# ========================================

print("⭐ AVIS MOYENS PAR PLATEFORME")
print("=" * 60)

# Joindre avec les avis
df_platform_ratings = df_clean.select(
    "name",
    "positive",
    "negative",
    col("platforms.windows").alias("windows"),
    col("platforms.mac").alias("mac"),
    col("platforms.linux").alias("linux")
).filter(
    (col("positive").isNotNull()) &
    (col("negative").isNotNull()) &
    ((col("positive") + col("negative")) > 0)
) \
    .withColumn("total_avis", col("positive") + col("negative")) \
    .withColumn("ratio_positif", (col("positive") / col("total_avis")) * 100)

# Ratio positif moyen par plateforme
print("🏆 RATIO POSITIF MOYEN :")

# Windows
windows_rating = df_platform_ratings.filter(col("windows") == True) \
    .select(
        round(avg("ratio_positif"), 2).alias("Ratio_positif_Windows"),
        round(avg("total_avis"), 0).alias("Nb_avis_moyen_Windows")
    )
display(windows_rating)

# Mac
mac_rating = df_platform_ratings.filter(col("mac") == True) \
    .select(
        round(avg("ratio_positif"), 2).alias("Ratio_positif_Mac"),
        round(avg("total_avis"), 0).alias("Nb_avis_moyen_Mac")
    )
display(mac_rating)

# Linux
linux_rating = df_platform_ratings.filter(col("linux") == True) \
    .select(
        round(avg("ratio_positif"), 2).alias("Ratio_positif_Linux"),
        round(avg("total_avis"), 0).alias("Nb_avis_moyen_Linux")
    )
display(linux_rating)

⭐ AVIS MOYENS PAR PLATEFORME
🏆 RATIO POSITIF MOYEN :


Ratio_positif_Windows,Nb_avis_moyen_Windows
73.66,1718.0


Ratio_positif_Mac,Nb_avis_moyen_Mac
77.7,3263.0


Ratio_positif_Linux,Nb_avis_moyen_Linux
78.28,3925.0


In [0]:
# ========================================
# 🎮 GENRES PAR PLATEFORME
# ========================================

print("🎮 GENRES LES PLUS REPRÉSENTÉS PAR PLATEFORME")
print("=" * 60)

# Créer un dataframe avec genres ET plateformes
df_genres_platforms = df_clean.select(
    "name",
    explode(split(col("genre"), ",")).alias("genre_simple"),
    col("platforms.windows").alias("windows"),
    col("platforms.mac").alias("mac"),
    col("platforms.linux").alias("linux")
) \
    .withColumn("genre_simple", trim(col("genre_simple"))) \
    .filter(col("genre_simple") != "")

# Top 10 genres sur Windows
print("\n🪟 TOP 10 GENRES SUR WINDOWS :")
windows_genres = df_genres_platforms.filter(col("windows") == True) \
    .groupBy("genre_simple") \
    .agg(count("*").alias("nombre_jeux")) \
    .orderBy(desc("nombre_jeux")) \
    .limit(10)

display(windows_genres)

# Top 10 genres sur Mac
print("\n🍎 TOP 10 GENRES SUR MAC :")
mac_genres = df_genres_platforms.filter(col("mac") == True) \
    .groupBy("genre_simple") \
    .agg(count("*").alias("nombre_jeux")) \
    .orderBy(desc("nombre_jeux")) \
    .limit(10)

display(mac_genres)

# Top 10 genres sur Linux
print("\n🐧 TOP 10 GENRES SUR LINUX :")
linux_genres = df_genres_platforms.filter(col("linux") == True) \
    .groupBy("genre_simple") \
    .agg(count("*").alias("nombre_jeux")) \
    .orderBy(desc("nombre_jeux")) \
    .limit(10)

display(linux_genres)

🎮 GENRES LES PLUS REPRÉSENTÉS PAR PLATEFORME

🪟 TOP 10 GENRES SUR WINDOWS :


genre_simple,nombre_jeux
Indie,39676
Action,23755
Casual,22082
Adventure,21427
Strategy,10892
Simulation,10832
RPG,9533
Early Access,6145
Free to Play,3391
Sports,2665


Databricks visualization. Run in Databricks to view.


🍎 TOP 10 GENRES SUR MAC :


genre_simple,nombre_jeux
Indie,9935
Casual,5130
Adventure,5039
Action,4564
Strategy,3005
Simulation,2439
RPG,2248
Early Access,900
Free to Play,845
Sports,506


Databricks visualization. Run in Databricks to view.


🐧 TOP 10 GENRES SUR LINUX :


genre_simple,nombre_jeux
Indie,6978
Action,3379
Casual,3305
Adventure,3302
Strategy,1826
Simulation,1532
RPG,1524
Early Access,632
Free to Play,474
Racing,304


Databricks visualization. Run in Databricks to view.

In [0]:
# ========================================
# 🎯 Y A-T-IL DES GENRES PRÉFÉRENTIELLEMENT SUR CERTAINES PLATEFORMES ?
# ========================================

print("🎯 ANALYSE DES PRÉFÉRENCES DE GENRES PAR PLATEFORME")
print("=" * 60)

# Calculer le % de jeux d'un genre disponibles sur chaque plateforme
genre_platform_preference = df_genres_platforms.groupBy("genre_simple") \
    .agg(
        count("*").alias("total_jeux_genre"),
        sum(when(col("windows") == True, 1).otherwise(0)).alias("sur_windows"),
        sum(when(col("mac") == True, 1).otherwise(0)).alias("sur_mac"),
        sum(when(col("linux") == True, 1).otherwise(0)).alias("sur_linux")
    ) \
    .filter(col("total_jeux_genre") >= 50) \
    .withColumn("pct_windows", round((col("sur_windows") / col("total_jeux_genre")) * 100, 1)) \
    .withColumn("pct_mac", round((col("sur_mac") / col("total_jeux_genre")) * 100, 1)) \
    .withColumn("pct_linux", round((col("sur_linux") / col("total_jeux_genre")) * 100, 1)) \
    .select("genre_simple", "total_jeux_genre", "pct_windows", "pct_mac", "pct_linux") \
    .orderBy(desc("total_jeux_genre"))

display(genre_platform_preference)

print("\n💡 Interprétation :")
print("- Si pct_windows est proche de 100% : genre très présent sur Windows")
print("- Si pct_mac ou pct_linux est faible : genre moins porté sur ces plateformes")

🎯 ANALYSE DES PRÉFÉRENCES DE GENRES PAR PLATEFORME


genre_simple,total_jeux_genre,pct_windows,pct_mac,pct_linux
Indie,39681,100.0,25.0,17.6
Action,23759,100.0,19.2,14.2
Casual,22086,100.0,23.2,15.0
Adventure,21431,100.0,23.5,15.4
Strategy,10895,100.0,27.6,16.8
Simulation,10836,100.0,22.5,14.1
RPG,9534,100.0,23.6,16.0
Early Access,6145,100.0,14.6,10.3
Free to Play,3393,99.9,24.9,14.0
Sports,2666,100.0,19.0,10.8



💡 Interprétation :
- Si pct_windows est proche de 100% : genre très présent sur Windows
- Si pct_mac ou pct_linux est faible : genre moins porté sur ces plateformes


In [0]:
# ========================================
# 📊 JEUX DISPONIBLES SUR TOUTES LES PLATEFORMES
# ========================================

print("📊 JEUX MULTI-PLATEFORMES LES PLUS POPULAIRES")
print("=" * 60)

# Filtrer les jeux sur les 3 plateformes
cross_platform_games = df_clean.select(
    "name",
    "positive",
    "negative",
    col("platforms.windows").alias("windows"),
    col("platforms.mac").alias("mac"),
    col("platforms.linux").alias("linux")
).filter(
    (col("windows") == True) &
    (col("mac") == True) &
    (col("linux") == True)
) \
    .filter(
        (col("positive").isNotNull()) &
        (col("negative").isNotNull())
    ) \
    .withColumn("total_avis", col("positive") + col("negative")) \
    .orderBy(desc("total_avis")) \
    .limit(20)

display(cross_platform_games)

print(f"\nNombre de jeux disponibles sur les 3 plateformes : {cross_platform_games.count()}")

📊 JEUX MULTI-PLATEFORMES LES PLUS POPULAIRES


name,positive,negative,windows,mac,linux,total_avis
Counter-Strike: Global Offensive,5943345,787093,true,true,true,6730438
Dota 2,1534895,317916,true,true,true,1852811
Terraria,1014711,22380,true,true,true,1037091
Team Fortress 2,846407,57423,true,true,true,903830
Garry's Mod,861240,29998,true,true,true,891238
Left 4 Dead 2,643836,16828,true,true,true,660664
Euro Truck Simulator 2,572368,15615,true,true,true,587983
Stardew Valley,497558,9283,true,true,true,506841
Unturned,451671,42098,true,true,true,493769
Don't Starve Together,340987,13943,true,true,true,354930



Nombre de jeux disponibles sur les 3 plateformes : 20


# 🎮 PROJET STEAM - ANALYSE GLOBALE DU MARCHÉ DES JEUX VIDÉO

- Analyse réalisée pour : UBISOFT
- Dataset : 55,691 jeux analysés sur Steam

## 📊 1. INSIGHTS MACRO - VUE D'ENSEMBLE DU MARCHÉ STEAM

### 🏢 TOP ÉDITEURS :
- Le marché Steam montre une concentration sur quelques gros éditeurs
- Les éditeurs indépendants sont très nombreux mais publient peu
- Opportunité pour Ubisoft : se positionner dans le top 10

### ⭐ QUALITÉ DES JEUX :
- Ratio positif moyen observé : ~75-85%
- Les jeux avec >90% d'avis positifs sont rares et très valorisés
- La qualité au lancement est CRUCIALE pour la réputation
- Les premiers avis (1ère semaine) impactent durablement les ventes

### 💰 DISTRIBUTION DES PRIX :
- Forte concentration de jeux gratuits (Free-to-Play)
- Jeux payants : majorité entre 5-20€
- AAA premium (40-70€) : segment minoritaire mais lucratif
- Prix moyen global relativement bas (tiré vers le bas par F2P et indie)

### 🎁 PROMOTIONS :
- Les promotions sont massivement utilisées sur Steam
- Steam Sales (été, hiver) : événements incontournables
- Stratégie recommandée : prix plein au lancement, puis promos ciblées

### 📊 CATÉGORIES FONCTIONNELLES :
- Single-player : catégorie dominante (>40k jeux)
- Multi-player : présent mais minoritaire (~10k jeux)
- Steam Achievements : très populaire (~25k jeux)
- Steam Cloud : fonctionnalité appréciée (~15k jeux)
- Steam Trading Cards : engagement communautaire
- Full controller support : important pour accessibilité
- => 💡 INSIGHT : Single-player domine, mais multi-player génère plus d'engagement

## 🎮 2. INSIGHTS GENRES - QUE NOUS DISENT LES DONNÉES ?

### 📊 TOP GENRES LES PLUS REPRÉSENTÉS :
1. Indie (~20,000+ jeux) - Marché très saturé
2. Action (~15,000 jeux) - Segment ultra-compétitif
3. Casual (~10,000 jeux) - Jeux légers, marché mobile/PC
4. Adventure (~10,000 jeux) - Narration et exploration
5. Strategy (~6,000 jeux) - Audience de niche fidèle
6. Simulation (~5,000 jeux) - Diversité (city builders, farming, etc.)
7. RPG (~5,000 jeux) - Audiences très engagées
- => ⚠️ Indie + Action = 60%+ du catalogue → concurrence féroce

### ⭐ GENRES AVEC MEILLEURS RATIOS POSITIFS (top performers) :
- Genres de niche avec communautés passionnées
- RPG et Strategy : audiences exigeantes mais fidèles
- Simulation : satisfaction élevée (gameplay sandbox)
- Sports : fans engagés de franchises
- => 💡 INSIGHT : Les genres de niche bien exécutés > genres populaires médiocres

### 💰 PRIX MOYEN PAR GENRE (observations) :
- Simulation et Strategy : prix moyens plus élevés (complexité)
- RPG : souvent premium (30-60€) si AAA, mais beaucoup d'indie à -15€
- Action : très variable (F2P → 70€)
- Indie : généralement -15€
- Sports : prix élevés pour franchises établies (FIFA, etc.)

### 🏢 GENRES PRÉFÉRÉS DES TOP ÉDITEURS :
- Les gros éditeurs diversifient mais gardent des spécialités
- Action-Adventure : combo gagnant pour AAA
- Les indie focalisent sur 1-2 genres max

### 🎯 RECOMMANDATIONS POUR UBISOFT :
- ✓ ACTION-ADVENTURE : zone de confort Ubisoft, marché prouvé
- ✓ Éviter pur 'Indie' positioning (Ubisoft = AAA)
- ✓ Intégrer éléments RPG pour profondeur (AC Valhalla modèle)
- ✓ Multi-joueur coopératif pour engagement long terme
- ✓ Simulation légère (base building, customisation) pour rétention
- ✓ SE DIFFÉRENCIER : Narrative forte, monde ouvert, polish technique

## 💻 3. INSIGHTS PLATEFORMES - DISTRIBUTION STEAM

### 📊 RÉPARTITION EXACTE :
- Windows : 55,676 jeux (99.97% du catalogue)
- Mac : 12,770 jeux (22.94% du catalogue)
- Linux : 8,458 jeux (15.19% du catalogue)
- => 🔑 INSIGHT CLÉ : Windows = quasi-monopole, Mac/Linux = 15-23%

### 🎮 ANALYSE MULTI-PLATEFORMES :
- ~15,000-20,000 jeux : disponibles sur les 3 plateformes (27-36%)
- ~15,000 jeux : disponibles sur 2 plateformes
- ~20,000+ jeux : exclusifs Windows (36%+)
- => 💡 INSIGHT : 1/3 des jeux sont Windows-only, 1/3 sont cross-platform

### 💰 PRIX MOYEN PAR PLATEFORME :
- Prix Windows : légèrement variable (grand volume)
- Prix Mac : tendance légèrement supérieure (audience premium)
- Prix Linux : similaire ou légèrement inférieur
- => 💡 Les jeux portés Mac/Linux sont souvent de meilleure qualité (biais)

### 🎮 TOP GENRES PAR PLATEFORME :
- WINDOWS :
  - Indie, Action, Casual dominent (même ordre que global)
  - Tous genres représentés
- MAC :
  - Indie (~5,000+ jeux) - Forte présence
  - Action (~3,500 jeux)
  - Casual (~2,500 jeux)
  - Strategy bien représentée (Mac = plateforme strategy historique)
- LINUX :
  - Indie (~3,500+ jeux) - Communauté open-source
  - Action (~2,000 jeux)
  - Strategy (~1,500 jeux)
  - RPG bien présent

### 🎯 RECOMMANDATIONS STRATÉGIQUES PLATEFORMES :
- 🥇 PRIORITÉ ABSOLUE - WINDOWS :
  - Optimisation maximale (99.97% du marché)
  - Support toutes configs (low-end → high-end)
  - Compatibilité Steam Deck (architecture Linux mais crucial)
- 🥈 PRIORITÉ SECONDAIRE - MAC :
  - ROI à évaluer : 23% du marché, audience plutôt premium
  - Si portage : prévoir dès la conception (pas après-coup)
  - Apple Silicon (M1/M2/M3) : architecture ARM, portage complexe
  - Public Mac : casual gamers + strategy fans
- 🥉 PRIORITÉ TERTIAIRE - LINUX :
  - 15% du marché, communauté passionnée et vocale
  - Si portage : via Proton/Wine validation OK pour démarrer
  - Steam Deck = Linux → validation Linux = validation Deck
  - Coût/bénéfice souvent limité (sauf jeux indie/strategy)

### ✅ STRATÉGIE UBISOFT RECOMMANDÉE :
1. Windows natif : 100% effort
2. Steam Deck : validation day-1 (marché émergent)
3. Mac : évaluation post-lancement (si succès Windows)
4. Linux : si Mac fait, Linux vient quasi-gratuitement

## 🚀 4. PLAN D'ACTION POUR LE NOUVEAU JEU UBISOFT

### 📋 SYNTHÈSE DES INSIGHTS CLÉS :
1. Marché Steam = 55,691 jeux → TRÈS COMPÉTITIF
2. Windows = 99.97% → Incontournable, optimisation maximale
3. Action + Indie = 60%+ du catalogue → Différenciation cruciale
4. Ratio positif moyen ~75-85% → Viser >85% pour se démarquer
5. Single-player domine (40k+) mais multi-player = engagement
6. Promotions massives → Prévoir stratégie prix long terme

### 🎯 RECOMMANDATION #1 - POSITIONNEMENT PRODUIT :
- GENRE RECOMMANDÉ : Action-Adventure + éléments RPG
- POURQUOI ?
  - ✓ Zone d'excellence Ubisoft (Far Cry, AC, Watch Dogs)
  - ✓ Marché prouvé et large (Action = 15k jeux mais demande énorme)
  - ✓ RPG elements = profondeur et rétention (tendance actuelle)
  - ✓ Permet narration forte (atout Ubisoft)
- DIFFÉRENCIATEURS CLÉS :
  - ✓ Monde ouvert de qualité AAA (expertise Ubisoft)
  - ✓ Narration cinématique (vs indie génériques)
  - ✓ Polish technique (top 15% qualité)
  - ✓ Contenu post-lancement planifié (DLC, seasons)
  - ✓ Éléments multi-joueur optionnels (co-op, raids)

### 💰 RECOMMANDATION #2 - STRATÉGIE PRIX :
- LANCEMENT : 59.99€ - 69.99€ (pricing AAA standard)
- JUSTIFICATION :
  - ✓ AAA Ubisoft = attente qualité premium
  - ✓ Budget marketing supporte prix élevé
  - ✓ Contenu riche justifie investissement
- ÉDITIONS :
  - ✓ Standard : 59.99€ (jeu de base)
  - ✓ Gold : 79.99€ (+ Season Pass)
  - ✓ Ultimate : 99.99€ (+ bonus cosmétiques, early access 3 jours)
POST-LANCEMENT :
  - ✓ Semaines 1-4 : prix plein (early adopters)
  - ✓ Mois 2-3 : -10% (première promo légère)
  - ✓ Steam Summer Sale : -25% (6 mois post-launch)
  - ✓ Steam Winter Sale : -33% (1 an post-launch)
  - ✓ Promotions régulières après an 1 : -50% à -75%
- => 💡 NE PAS brader trop vite → dévalue la marque

### 💻 RECOMMANDATION #3 - PLATEFORMES :
- PHASE 1 - LANCEMENT (Jour 1) :
  - ✅ Windows (toutes configs : low, medium, high, ultra)
  - ✅ Steam Deck verification (via Proton)
- PHASE 2 - POST-LANCEMENT (3-6 mois) :
  - 🔄 Mac (si succès commercial Windows)
     - → Évaluation : coût portage vs 23% marché
     - → Si portage : M1/M2/M3 prioritaire
- PHASE 3 - EXTENSION (6-12 mois) :
  - 🔄 Linux natif (si Mac fait)
     - → Vient presque gratuitement après Mac
     - → Bonus : améliore compatibilité Steam Deck
  - ⚠️ ATTENTION : Annoncer intentions multi-plateforme AVANT lancement
     - → Communauté Mac/Linux apprécie transparence
     - → Éviter backlash "Windows only"

### 🎮 RECOMMANDATION #4 - FONCTIONNALITÉS STEAM :
- OBLIGATOIRE (attendu par communauté) :
  - ✅ Steam Achievements (~50 achievements variés)
  - ✅ Steam Cloud (save syncing)
  - ✅ Full controller support (manette Xbox/PS)
  - ✅ Steam Trading Cards (engagement casual)
  - ✅ Steam Leaderboards (si éléments compétitifs)
- RECOMMANDÉ (valeur ajoutée) :
  - ✅ Steam Workshop (si modding pertinent)
  - ✅ Steam Remote Play Together (co-op local→online)
  - ✅ Steam Friends integration
  - ✅ Twitch/Discord Rich Presence  
- OPTIONNEL (selon gameplay) :
  - 🔄 Steam Inventory (items tradables - si pertinent)
  - 🔄 In-game economy avec Steam Market  
- => 💡 Ces features = différenciateur vs concurrence lazy

### ⭐ RECOMMANDATION #5 - STRATÉGIE QUALITÉ :
- OBJECTIF : Ratio avis positifs >85% (top 15% du marché)
- PRÉ-LANCEMENT :
  - ✓ Beta testing étendue (6+ mois avant launch)
  - ✓ Programme early access sélectif (influenceurs, fans)
  - ✓ Polish technique ABSOLU (zéro tolérance bugs critiques)
  - ✓ Optimisation performance (60 FPS stable sur configs moyennes)
- LANCEMENT :
  - ✓ Support day-1 réactif (hotfix -24h si crash)
  - ✓ Community managers actifs sur forums Steam
  - ✓ Transparence totale sur roadmap post-launch
- POST-LANCEMENT :
  - ✓ Patches réguliers (mensuel minimum)
  - ✓ Écoute feedback communauté
  - ✓ Amélioration continue QoL (quality of life)
- => ⚠️ CRITIQUE : Premiers avis (48h) = 80% de l'impact réputation → Launch DOIT être impeccable